src/ObjectUtils/ElfFile.cpp

```diff
@@ -100,7 +100,10 @@ class ElfFileImpl : public ElfFile {
   ErrorMessageOr<void> InitSections();
   ErrorMessageOr<void> InitProgramHeaders();
   ErrorMessageOr<void> InitDynamicEntries();
-  ErrorMessageOr<SymbolInfo> CreateSymbolInfo(const llvm::object::ELFSymbolRef& symbol_ref);
+  ErrorMessageOr<SymbolInfo> CreateSymbolInfo(
+      const llvm::object::ELFSymbolRef& symbol_ref,
+      const absl::flat_hash_set<uint64_t>& hotpachable_addresses);
+  [[nodiscard]] absl::flat_hash_set<uint64_t> LoadHotpatchableAddresses();
```

Comment:
What do you think about having the hotpatchable_addresses_, as optional member that gets filled on the LoadHotpatchableAddresses function? I don't have a strong opinion here, though.

Reply:
Do you know how large the memory footprint would be?
I mean we do have already 2-3 calls to this parsing for a single symbol loading.

But you are probably right. Consider adding a brief comment.

Reply:
> memory footprint
Not that much; an eight byte address for every symbol plus overhead. I'll just leave it as is.